tests/test_decomposers.py

```diff
@@ -383,6 +383,28 @@ def test_next_on_different_sized_constraints(self):
         self.assertIn('subproblem', newstate)
         self.assertTrue(len(newstate.subproblem) <= size)  # correct size
 
+    def test_partially_disconnected_constraints(self):
+        bqm = dimod.BinaryQuadraticModel.from_ising({}, {'ab': 1, 'bc': 1, 'ca': 1})
+        size = 2
+        constraints = constraints = ['a', 'b', 'cb']
+
+        rcd = RandomConstraintDecomposer(size, constraints)
+        state = State.from_problem(bqm)
+        newstate = rcd.run(state).result()
+
+        self.assertIn('subproblem', newstate)
+        self.assertTrue(len(newstate.subproblem) <= size)  # correct size
+
+    def test_completely_disconnected_constraints(self):
+        bqm = dimod.BinaryQuadraticModel.from_ising({}, {'ab': 1, 'bc': 1, 'ca': 1})
+        size = 2
+        constraints = constraints = ['a', 'b']
```

Comment:
Why are you assigning `constraints` twice?

Reply:
Got no idea how that slipped in. Thanks.